src/java/org/apache/cassandra/db/virtual/AccordDebugKeyspace.java

```diff
@@ -0,0 +1,671 @@
+                          .column("shard_only_applied_or_invalidated_before", decompose(entry.shardOnlyAppliedOrInvalidatedBefore))
+                          .column("bootstrapped_at", decompose(entry.bootstrappedAt))
+                          .column("stale_until_at_least", entry.staleUntilAtLeast != null ? decompose(entry.staleUntilAtLeast) : null);
+                        return ds;
+                    },
+                    dataSet,
+                    ignore -> false
+                );
+            }
+            return dataSet;
+        }
+    }
+
+    public static final class RejectBeforeTable extends AbstractVirtualTable
+    {
+        private RejectBeforeTable(String keyspace)
+        {
+            super(parse(keyspace,
+                        "Accord per-CommandStore RejectBefore State",
+                        "CREATE TABLE reject_before (\n" +
+                        "  command_store_id int,\n" +
+                           format("range_start %s,\n", ROUTING_KEY_TYPE_STRING) +
+                           format("range_end %s,\n", ROUTING_KEY_TYPE_STRING) +
+                           format("txn_id %s,\n", TXN_ID_TYPE_STRING) +
+                        "  PRIMARY KEY (command_store_id, range_start, range_end)" +
+                        ')'));
+        }
+
+        @Override
+        public DataSet data()
+        {
+            CommandStores stores = ((AccordService) AccordService.instance()).node().commandStores();
+            List<Pair<Integer, RejectBefore>> rangeMaps =
+                getBlockingAndRethrow(stores.map(store -> Pair.create(store.commandStore().id(), store.commandStore().unsafeGetRejectBefore())));
+            rangeMaps.sort(comparing(p -> p.left));
+
+            SimpleDataSet dataSet = new SimpleDataSet(metadata());
+            for (Pair<Integer, RejectBefore> pair : rangeMaps)
+            {
+                int storeId = pair.left;
+                RejectBefore rejectBefore = pair.right;
+
+                if (rejectBefore == null)
+                    continue;
+
+                rejectBefore.foldlWithBounds(
+                    (txnId, ds, start, end) -> ds.row(storeId, decompose(start), decompose(end)).column("txn_id", decompose(txnId)),
+                    dataSet,
+                    ignore -> false
+                );
+            }
+            return dataSet;
+        }
+    }
+
+    public static class TxnBlockedByTable extends AbstractVirtualTable
+    {
+        enum Reason { Self, Txn, Key }
+
+        protected TxnBlockedByTable(String keyspace)
+        {
+            super(parse(keyspace,
+                        "Accord Transactions Blocked By Table" ,
+                        "CREATE TABLE txn_blocked_by (\n" +
+                           format("txn_id %s,\n", TXN_ID_TYPE_STRING) +
+                        "  command_store_id int,\n" +
+                        "  depth int,\n" +
+                           format("blocked_by %s,\n", TXN_ID_TYPE_STRING) +
```

Comment:
same as above, why are we moving away from the string version that gets reported to users/operators?

Reply:
Addressed.